ambry-named-mysql/src/main/java/com/github/ambry/named/MySqlNamedBlobDb.java

```diff
@@ -124,6 +124,20 @@ class MySqlNamedBlobDb implements NamedBlobDb {
           + "WHERE t1.blob_name LIKE ? AND t1.blob_name >= ? ORDER BY t1.blob_name ASC LIMIT ?",STATE_MATCH, CURRENT_TIME);
   // @formatter:on
 
+  /**
+   * Similar like LIST_QUERY_V2, but this query is used when user don't provide the prefix and want to list all records.
+   */
+  // @formatter:off
+  private static final String LIST_ALL_QUERY_V2 = String.format(
+      "" + "SELECT t1.blob_name, t1.blob_id, t1.version, t1.deleted_ts " + "FROM named_blobs_v2 t1 " + "INNER JOIN "
+          + "(SELECT account_id, container_id, blob_name, max(version) as version " + "FROM named_blobs_v2 "
+          + "WHERE (account_id, container_id) = (?, ?) AND %1$s " + "  AND (deleted_ts IS NULL OR deleted_ts>%2$S) "
+          + "        GROUP BY account_id, container_id, blob_name) t2 "
+          + "ON (t1.account_id,t1.container_id,t1.blob_name,t1.version) = (t2.account_id,t2.container_id,t2.blob_name,t2.version) "
+          + "WHERE CASE WHEN ? IS NOT NULL THEN t1.blob_name >= ? ELSE 1 END ORDER BY t1.blob_name ASC LIMIT ?",
```

Comment:
Let's break this where case when then statement into multiple lines for readibility.

Reply:
Updated.